Source/WebCore/Modules/compression/CompressionStreamEncoder.h

```diff
@@ -47,14 +50,21 @@ class CompressionStreamEncoder : public RefCounted<CompressionStreamEncoder> {
 
     ~CompressionStreamEncoder()
     {
-        if (m_initialized)
-            deflateEnd(&m_zstream);
+        if (m_initialized) {
+            if (m_usingAppleCompressionFramework) {
+#if PLATFORM(COCOA)
+                compression_stream_destroy(&m_stream);
```

Comment:
It would be nice if we had a smart-pointer-like thing that would take care of calling compression_stream_destroy at the proper time.